commands/auth.go

```diff
@@ -71,187 +63,39 @@ func Auth() *Command {
-	if err != nil {
-		return err
-	}
-
-	viper.Set("access-token", token)
-
-	fmt.Println("updated access token")
-
-	return nil
-}
-
-type doitServerAuth struct {
-	url         string
-	browserOpen func(u string) error
-	isCLI       func() bool
-	monitorAuth func(u string, ac *doitserver.AuthCredentials) (*doitserver.TokenResponse, error)
-}
-
-func newDoitServerAuth() *doitServerAuth {
-	return &doitServerAuth{
-		url: "http://doit-server.apps.pifft.com",
-		browserOpen: func(u string) error {
-			return webbrowser.Open(u, webbrowser.NewTab, true)
-		},
-		isCLI: func() bool {
-			return (runtime.GOOS == "linux" && os.Getenv("DISPLAY") == "") || os.Getenv("CLIAUTH") != ""
-		},
-		monitorAuth: monitorAuthWS,
-	}
-}
-
-func (dsa *doitServerAuth) initAuth(ac *doitserver.AuthCredentials) (string, error) {
-	if dsa.isCLI() {
-		return dsa.initAuthCLI(ac)
-	}
-
-	return dsa.initAuthBrowser(ac)
-}
-
-func (dsa *doitServerAuth) initAuthCLI(ac *doitserver.AuthCredentials) (string, error) {
-	u, err := dsa.createAuthURL(ac, keyPair{k: "cliauth", v: "1"})
-	if err != nil {
-		return "", err
-	}
-
-	fmt.Printf("Visit the following URL in your browser: %s\n", u)
-
-	return retrieveUserTokenFunc()
-}
-
-func (dsa *doitServerAuth) initAuthBrowser(ac *doitserver.AuthCredentials) (string, error) {
-	u, err := dsa.createAuthURL(ac)
-	if err != nil {
-		return "", err
-	}
-
-	err = dsa.browserOpen(u)
-	if err != nil {
-		return "", err
-	}
-
-	tr, err := dsa.monitorAuth(dsa.url, ac)
-	if err != nil {
-		return "", err
-	}
-
-	return tr.AccessToken, nil
-}
-
-func (dsa *doitServerAuth) retrieveAuthCredentials() (*doitserver.AuthCredentials, error) {
-	u, err := url.Parse(dsa.url)
-	if err != nil {
-		return nil, err
-	}
-
-	u.Path = "/token"
-	v := u.Query()
-	v.Set("id", uuid.NewV4().String())
-	u.RawQuery = v.Encode()
-
-	r, err := http.Get(u.String())
-	if err != nil {
-		return nil, err
-	}
-	defer r.Body.Close()
-	if err != nil {
-		return nil, err
-	}
-
-	body, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		return nil, err
-	}
-
-	if r.StatusCode != 200 {
-		return nil, errors.New("it's broke, Jim")
-	}
-
-	var m doitserver.AuthCredentials
-	err = json.Unmarshal(body, &m)
-	if err != nil {
-		return nil, err
-	}
-
-	return &m, nil
-}
-
-type keyPair struct {
-	k, v string
-}
-
-func (dsa *doitServerAuth) createAuthURL(ac *doitserver.AuthCredentials, kps ...keyPair) (string, error) {
-	authURL, err := url.Parse(dsa.url)
+// RunAuthInit initializes the doctl config. Configuring is stored in $XDG_CONFIG_HOME/doctl. On Unix, if
+// XDG_CONFIG_HOME is not set, use $HOME/.config. On Windows use %APPDATA%/doctl/config.
+func RunAuthInit(c *CmdConfig) error {
+	in, err := retrieveUserTokenFunc()
 	if err != nil {
-		return "", err
-	}
-
-	authURL.Path = "/auth/digitalocean"
-
-	q := authURL.Query()
-	q.Set("id", ac.ID)
-	q.Set("cs", ac.CS)
-
-	for _, kp := range kps {
-		q.Set(kp.k, kp.v)
+		return errors.New("unable to read input")
```

Comment:
should include the error, and this scope doesn't know about _user input_, it knows about _user token_ so maybe the message should be like:
```go
return fmt.Errorf("unable to obtain user token: %v", err)
```